docs/resource_dependencies.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Resource Dependencies
+layout: docs
+---
+
+Say you have a web service specified in a Tilt resource named `frontend` and a
+database in a Tilt resource named `database`, and `frontend` fails to even
+start up if `database` isn't running. This can lead to distracting errors on
+startup (especially, e.g., if you have 5 services all depending on the same
+backend!)
+
+Tilt allows you to specify that `frontend` depends on `database`, so that
+`frontend` will not be deployed until `database` has been successfully deployed:
+
+```python
+k8s_resource('frontend', resource_deps=['database'])
+```
+
+This has two effects:
+1. `frontend` will not be deployed until `database` has been ready at least once
+    since Tilt was started.
+2. If you run `tilt up frontend` to select only some of your Tiltfile's resources,
+    that also implicitly selects all of `frontend`'s transitive dependencies.
+
+A resource is "ready" when:
+* For K8s resources: the pod is running and K8s considers all of its containers ready
+* For docker-compose resources: the container is started (NB: Tilt doesn't currently observe docker-compose health checks)
+* For local resources: the command has succeeded at least once
+
+Some other use cases:
+* Define your resource deps such that teammates who work on different subsets of services
```

Comment:
`tilt up svc1 svc2` looks the long version, not the short version--i bet there's a way to reword this so it's punchier and gets the point across better (e.g. "`tilt up frontend` brings up frontend, the database, and the asset server -- everything someone needs to develop on `frontend`, without needing to explicitly list the deps.)